src/legacy/ui/public/vis/editors/default/default_editor_utils.tsx

```diff
@@ -17,56 +17,59 @@
  * under the License.
  */
 
-import { EuiComboBoxOptionProps } from '@elastic/eui';
-import { AggType } from 'ui/agg_types';
+interface ComboBoxOption<T> {
+  label: string;
+  target: T;
+}
 
 // NOTE: we cannot export the interface with export { InterfaceName }
 // as there is currently a bug on babel typescript transform plugin for it
 // https://github.com/babel/babel/issues/7641
 //
-export type ComboBoxGroupedOption = EuiComboBoxOptionProps & {
-  label?: string;
-  value?: AggType;
-  options?: ComboBoxGroupedOption[];
-};
+export interface ComboBoxGroupedOption<T> {
+  label: string;
+  target?: T;
```

Comment:
I renamed `value` to `target` because according to `EUIComboBox` type `value` can be a simple type only `string | number | string[] | undefined` (https://github.com/elastic/eui/blob/master/src/components/combo_box/index.d.ts#L20)